pallets/author-noting/src/benchmarks.rs

```diff
@@ -57,31 +57,45 @@ benchmarks! {
         // Depend on the number of parachains registered
         let x in 0..100;
 
-        let mut sproof_builder = test_sproof::ParaHeaderSproofBuilder::default();
         let mut container_chains = vec![];
 
-        for para_id in 0..x {
-            let para_id = para_id.into();
-            container_chains.push(para_id);
-            // Mock assigned authors for this para id
-            let author: T::AccountId = account("account id", 0u32, 0u32);
-            // Use the max allowed value for num_each_container_chain
-            let num_each_container_chain = 2;
-            T::ContainerChainAuthor::set_authors_for_para_id(para_id, vec![author; num_each_container_chain]);
-            sproof_builder.num_items += 1;
-        }
-
-        let (root, proof) = sproof_builder.into_state_root_and_proof();
-
-        let data = tp_author_noting_inherent::OwnParachainInherentData {
-            relay_storage_proof: proof,
+        let data = if TypeId::of::<<<T as Config>::RelayOrPara as RelayOrPara>::InherentArg>() == TypeId::of::<tp_author_noting_inherent::OwnParachainInherentData>() {
```

Comment:
Can we create something like `InherentDataOf<T>`?

Reply:
Why are we comparing `TypeId`, would not be better to have a trait restriction for now with a note?

Reply:
Sorry I missed this, using TypeId is a hack but it was the fastest way I could think of to get this working. It is mainly used to get the code to compile. If I added a where bound forcing those 2 types to be equal this would not compile with the starlight runtime, only with dancebox and flashbox. And I didn't want to add a benchmark-only trait method like we usually do because this one depends on the mocked proof, so I would need to refactor that as well.

Anyway when we start running benchmarks for starlight we will need to think of a better way of handling the mock proof, so we will probably remove this hack as well.